### PR TITLE
Trev8939/dictionary renderer web style

### PR DESCRIPTION
**Pull request by TADraeseke**
PR: https://devtopia.esri.com/runtime/common-samples/issues/2326

### CODE REVIEW

File: kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt [hunk 1]
```diff
@@ -6,58 +6,83 @@ import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
...
+        val dictionaryRendererFromPortal = DictionaryRenderer(dictionarySymbolStyleFromPortal)
+
+        // on web style click
+        webStyleRadioButton.setOnClickListener {
+            // set the feature layer renderer to the dictionary renderer from portal
+            featureLayer.renderer = dictionaryRendererFromPortal
+        }
+    }
+
+
```
Comment (yo1995) [inline kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt:73]:
```suggestion
```

Don't know if Android uses a linter? 😛

File: kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt [hunk 2]
```diff
@@ -6,58 +6,83 @@ import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
...
+        // location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // create a feature layer from a service feature table
+        val featureTable =
+            ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Redlands_Restaurants/FeatureServer/0")
+        val featureLayer = FeatureLayer(featureTable)
+
+        // create a new map with a streets basemap and set it to the map view
+        mapView.map = ArcGISMap(BasemapStyle.ARCGIS_STREETS).apply {
```
Comment (yo1995) [inline kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt:34]:
```suggestion
        mapView.map = ArcGISMap(BasemapStyle. ARCGIS_TOPOGRAPHIC).apply {
```
Also the screenshot needs to be replaced.

File: kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt [hunk 3]
```diff
@@ -6,58 +6,83 @@ import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
...
+        // create a portal item using the portal and the item id of the dictionary web style
+        val portal = Portal("https://arcgisruntime.maps.arcgis.com")
+        val portalItem = PortalItem(portal, "adee951477014ec68d7cf0ea0579c800")
+        // map the input fields in the feature layer to the dictionary symbol style's expected fields for symbols and text
+        val fieldMap: HashMap<String, String> = HashMap()
+        fieldMap["healthgrade"] = "Inspection"
+        // create a new dictionary symbol style from the web style in the portal item
+        val dictionarySymbolStyleFromPortal = DictionarySymbolStyle(portalItem)
+        // create a new dictionary renderer from the dictionary symbol style
+        val dictionaryRendererFromPortal = DictionaryRenderer(dictionarySymbolStyleFromPortal)
```
Comment (yo1995) [inline kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt:64]:
```suggestion
        val dictionaryRendererFromPortal = DictionaryRenderer(dictionarySymbolStyleFromPortal, fieldMap, HashMap())
```
The fieldMap hasn't been applied... 😅 

Also, the screenshot needs update.

File: kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt [hunk 4]
```diff
@@ -6,58 +6,83 @@ import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
...
+        // on style file click
+        styleFileRadioButton.setOnClickListener {
+            // set the feature layer renderer to the dictionary renderer from local stylx file
+            featureLayer.renderer = dictionaryRendererFromFile
+        }
+        // set the initial state to use the dictionary renderer from local stylx file
+        styleFileRadioButton.performClick()
+
+        // create a portal item using the portal and the item id of the dictionary web style
+        val portal = Portal("https://arcgisruntime.maps.arcgis.com")
```
Comment (yo1995) [inline kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt:56]:
```suggestion
        val portal = Portal("https://arcgis.com")
```
Feel free to ignore this suggestion if it doesn't apply on Android.

On iOS, we have a convenient class method [`.arcGISOnline(withLoginRequired: false)`](https://developers.arcgis.com/ios/api-reference/interface_a_g_s_portal.html#a68398c03578deb91c307a96b8371a60d) so we typically don't use a string/URL literal. And generally we tend to cut off the prefix of the portal URL (i.e. the `arcgisruntime.maps.` portion).

**Review by yo1995 — COMMENTED** [review]
I didn't test out the local file part because I don't quite know where to put the file 😅 . And I didn't review the UI layout xml as the sample looks clean.

Other than a few small things, it looks good overall!

**Comment by yo1995** [comment]
Also for iOS we don't put the devtopia link in the public repo. Does it matter? It would be really convenient if we can paste that link on GitHub. See https://github.com/Esri/arcgis-runtime-samples-ios/pull/1047

File: kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt (hunk 4 above)
Comment (TADraeseke) [reply kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt:56]: Oh cool. We don't have a convenience method like that!

**Comment by TADraeseke** [comment]
Thanks @yo1995 - Made those fixes!

**Review by yo1995 — APPROVED** [review]

**Comment by yo1995** [comment]
> PR: https://devtopia.esri.com/runtime/common-samples/issues/2326

Don't want to be picky, but chatted with Phil and Divesh and we felt it might be better not to directly post internal URLs on GitHub - it could expose some potential attack surface... Just a thought 😛

**Pull request merged**